components/camel-zendesk/src/main/java/org/apache/camel/component/zendesk/ZendeskEndpoint.java

```diff
@@ -0,0 +1,115 @@
+import org.apache.camel.Producer;
+import org.apache.camel.component.zendesk.internal.ZendeskApiCollection;
+import org.apache.camel.component.zendesk.internal.ZendeskApiName;
+import org.apache.camel.component.zendesk.internal.ZendeskConstants;
+import org.apache.camel.component.zendesk.internal.ZendeskHelper;
+import org.apache.camel.component.zendesk.internal.ZendeskPropertiesHelper;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.util.ObjectHelper;
+import org.apache.camel.util.component.AbstractApiEndpoint;
+import org.apache.camel.util.component.ApiMethod;
+import org.apache.camel.util.component.ApiMethodPropertiesHelper;
+import org.zendesk.client.v2.Zendesk;
+
+/**
+ * The zendesk endpoint interacts with the Zendesk server.
+ */
+@UriEndpoint(firstVersion = "2.19.0", scheme = "zendesk", title = "Zendesk", syntax = "zendesk:methodName",
+    consumerClass = ZendeskConsumer.class, consumerPrefix = "consumer", label = "api,support,cloud", lenientProperties = true)
+public class ZendeskEndpoint extends AbstractApiEndpoint<ZendeskApiName, ZendeskConfiguration> {
+
+    @UriParam
+    private ZendeskConfiguration configuration;
+
+    private Zendesk apiProxy;
+
+    public ZendeskEndpoint(String uri, ZendeskComponent component, ZendeskApiName apiName, String methodName,
+                       ZendeskConfiguration endpointConfiguration) {
+        super(uri, component, apiName, methodName, ZendeskApiCollection.getCollection().getHelper(apiName),
+                endpointConfiguration);
+        this.configuration = endpointConfiguration;
+    }
+
+    @Override
+    public ZendeskComponent getComponent() {
+        return (ZendeskComponent) super.getComponent();
+    }
+
+    public Producer createProducer() throws Exception {
+        return new ZendeskProducer(this);
+    }
+
+    public Consumer createConsumer(Processor processor) throws Exception {
+        // make sure inBody is not set for consumers
+        if (inBody != null) {
+            throw new IllegalArgumentException("Option inBody is not supported for consumer endpoint");
+        }
+        final ZendeskConsumer consumer = new ZendeskConsumer(this, processor);
+        // also set consumer.* properties
+        configureConsumer(consumer);
+        return consumer;
+    }
+
+    @Override
+    public void doStop() throws Exception {
+        if (apiProxy != null && !apiProxy.isClosed()) {
+            apiProxy.close();
+        }
+        super.doStop();
+    }
+
+    @Override
+    protected ApiMethodPropertiesHelper<ZendeskConfiguration> getPropertiesHelper() {
+        return ZendeskPropertiesHelper.getHelper();
+    }
+
+    protected String getThreadProfileName() {
+        return ZendeskConstants.THREAD_PROFILE_NAME;
+    }
+
+    @Override
+    protected void afterConfigureProperties() {
```

Comment:
I'm not really sure if there's any big difference. I moved it to doStop() but this method still should be there as it's abstract method come from AbstractApiEndpoint.